crates/sui-framework-build/src/compiled_package.rs

```diff
@@ -0,0 +1,273 @@
+                        });
+                    }
+                }
+                if self.run_bytecode_verifier {
+                    for m in compiled_modules.iter_modules() {
+                        move_bytecode_verifier::verify_module(m).map_err(|err| {
+                            SuiError::ModuleVerificationFailure {
+                                error: err.to_string(),
+                            }
+                        })?;
+                        sui_bytecode_verifier::verify_module(m)?;
+                    }
+                    // TODO(https://github.com/MystenLabs/sui/issues/69): Run Move linker
+                }
+                Ok(CompiledPackage { package, path })
+            }
+        }
+    }
+}
+
+impl CompiledPackage {
+    /// Return all of the bytecode modules in this package.
+    /// Note: these are not topolically sorted by dependency--use `get_dependency_sorted_modules` to produce a list of modules suitable
+    /// for publishing or static analysis
+    pub fn get_modules(&self) -> Vec<&CompiledModule> {
+        self.package
+            .root_modules()
+            .map(|m| match &m.unit {
+                CompiledUnitEnum::Module(m) => &m.module,
+                CompiledUnitEnum::Script(_) => unimplemented!("Scripts not supported in Sui Move"),
+            })
+            .collect()
+    }
+
+    /// Return all of the bytecode modules that this package depends on (both directly and transitively)
+    /// Note: these are not topolically sorted by dependency-
+    pub fn get_dependent_modules(&self) -> Vec<&CompiledModule> {
+        self.package
+            .deps_compiled_units
+            .iter()
+            .map(|(_, m)| match &m.unit {
+                CompiledUnitEnum::Module(m) => &m.module,
+                CompiledUnitEnum::Script(_) => unimplemented!("Scripts not supported in Sui Move"),
+            })
+            .collect()
+    }
+
+    /// Return the bytecode modules in this package, topologically sorted in dependency order
+    /// This is the function to call if you would like to publish or statically analyze the modules
+    pub fn get_dependency_sorted_modules(&self) -> SuiResult<Vec<CompiledModule>> {
+        let compiled_modules = self.package.root_modules_map();
+        // Collect all module IDs from the current package to be
+        // published (module names are not sufficient as we may
+        // have modules with the same names in user code and in
+        // Sui framework which would result in the latter being
+        // pulled into a set of modules to be published).
+        // For each transitive dependent module, if they are not to be published,
+        // they must have a non-zero address (meaning they are already published on-chain).
+        let self_modules: HashSet<ModuleId> = compiled_modules
+            .iter_modules()
+            .iter()
+            .map(|m| m.self_id())
+            .collect();
+        if let Some(m) = self
```

Comment:
might be easier to read of it was bound to a variable beforehand, e.g. `let bad_module = self.package...; if let Some(m) = bad_module`